dumbbells/dataset.py

```diff
@@ -119,4 +136,38 @@ def reset(self):
         result = self.env.reset()
         # Update the current state space
         self.state_space = result[0]
+        # Reset our frames
+        self.frames = []
         return result
+
+    def save_frames_as_gif(self, path="./", filename="gym_animation.gif"):
+        """Produces a gif of the every action that has been taken by the agent on the environment from the
+        beginning until called
+
+        Args:
+            path: Optional, file path where the gif will be saved relative to the directory calling this method.
+                Default is the same directory as the caller.
+
+            filename: Optional, file name of the gif to be saved. Default is "gym_animation.gif"
+
+        Returns:
+            None (though it produces a saved gif)
+
+        """
+        self.env.close()
+        # Mess with this to change frame size
+        plt.figure(
+            figsize=(self.frames[0].shape[1] / 72.0, self.frames[0].shape[0] / 72.0),
+            dpi=72,
+        )
+
+        patch = plt.imshow(self.frames[0])
+        plt.axis("off")
+
+        def animate(i):
+            patch.set_data(self.frames[i])
```

Comment:
For a one-liner function like this you could use a lambda: `animate = lambda i: patch.set_data(self.frames[i])`

Or even include in in the animation block:
```
anim = animation.FuncAnimation(
     plt.gcf(), 
     lambda i: patch.set_data(self.frames[i]), 
     frames=len(self.frames), 
     interval=50
)
```